.travis.yml

```diff
@@ -60,46 +60,73 @@ jobs:
 
         - &STANDARD_TEST_JOB
             stage: Fast Test
-            php: 7.0
+            php: '7.2'
             install:
                 # Composer: enforce given Symfony components version
                 - if [ "$SYMFONY_VERSION" != "" ]; then composer global show symfony/flex -q || travis_retry composer global require $DEFAULT_COMPOSER_FLAGS symfony/flex; fi
                 - if [ "$SYMFONY_VERSION" != "" ]; then composer config extra.symfony.require $SYMFONY_VERSION || true; fi
 
-                - travis_retry composer update $DEFAULT_COMPOSER_FLAGS $COMPOSER_FLAGS
+                - travis_retry composer update --working-dir=./dev-tools $DEFAULT_COMPOSER_FLAGS $COMPOSER_FLAGS
                 - composer info -D | sort
             script:
-                - vendor/bin/phpunit || travis_terminate 1
+                - ./dev-tools/vendor/bin/phpunit || travis_terminate 1
                 - PHP_CS_FIXER_FUTURE_MODE=1 php php-cs-fixer --diff --dry-run -v fix
 
         -
             <<: *STANDARD_TEST_JOB
             stage: Test
-            php: 5.6
+            php: '5.6'
             env: COMPOSER_FLAGS="--prefer-stable --prefer-lowest"
+            before_install:
+                # turn off Xdebug
+                - phpenv config-rm xdebug.ini || return 0
+
+                # Composer: boost installation
+                - composer global show hirak/prestissimo --quiet || travis_retry composer global require $DEFAULT_COMPOSER_FLAGS hirak/prestissimo
+
+                # Require PHPUnit 8
+                - composer remove --dev --no-update --working-dir=./dev-tools humbug/box localheinz/composer-normalize phpstan/phpstan-phpunit
 
         -
             <<: *STANDARD_TEST_JOB
             stage: Test
-            php: 7.1
+            php: '7.0'
             env: SYMFONY_DEPRECATIONS_HELPER=disabled PHP_CS_FIXER_TEST_USE_LEGACY_TOKENIZER=1 SYMFONY_VERSION="~4.1.0"
+            before_install:
+                # turn off Xdebug
+                - phpenv config-rm xdebug.ini || return 0
+
+                # Composer: boost installation
+                - composer global show hirak/prestissimo --quiet || travis_retry composer global require $DEFAULT_COMPOSER_FLAGS hirak/prestissimo
+
+                # Require PHPUnit 8
+                - composer remove --dev --no-update --working-dir=./dev-tools humbug/box localheinz/composer-normalize phpstan/phpstan-phpunit
 
         -
             <<: *STANDARD_TEST_JOB
             stage: Test
-            php: 7.2
+            php: '7.1'
+            before_install:
+                # turn off Xdebug
+                - phpenv config-rm xdebug.ini || return 0
+
+                # Composer: boost installation
+                - composer global show hirak/prestissimo --quiet || travis_retry composer global require $DEFAULT_COMPOSER_FLAGS hirak/prestissimo
```

Comment:
DRY is violated

Reply:
I didn't know there is a rule to not repeat the code more then twice ;)

Do you know how can we fix this and currently existing duplication as well?

Reply:
probably each duplication to discuss separately. i'm talking about multiple places in travis file when you added same step over and over ago. not needed, as proven on master branch already.